From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Tue, 3 Mar 2015 17:10:38 +0800
Subject: [PATCH] {BUG?/WIP} drivers/bluetooth/dtl1_cs: comment an always-true
 part of a or comparaison. Using && instead of || might do what's expected?

Reported: https://lkml.org/lkml/2017/1/15/44
---
 drivers/bluetooth/dtl1_cs.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/bluetooth/dtl1_cs.c b/drivers/bluetooth/dtl1_cs.c
index 6317c6f323bf..69c48f1b9b1d 100644
--- a/drivers/bluetooth/dtl1_cs.c
+++ b/drivers/bluetooth/dtl1_cs.c
@@ -556,7 +556,10 @@ static void dtl1_detach(struct pcmcia_device *link)
 
 static int dtl1_confcheck(struct pcmcia_device *p_dev, void *priv_data)
 {
-	if ((p_dev->resource[1]->end) || (p_dev->resource[1]->end < 8))
+	/* Why is the first member without a comparaison?
+	 * Introduced by 00990e7ce0b0, found with -Wlogical-op
+	 */
+	if ((p_dev->resource[1]->end)/* || (p_dev->resource[1]->end < 8)*/)
 		return -ENODEV;
 
 	p_dev->resource[0]->flags &= ~IO_DATA_PATH_WIDTH;
-- 
